File: C#/240. Search a 2D Matrix II.cs
public class Solution
{
    public bool SearchMatrix(int[,] matrix, int target)
    {
        int row = 0, col = matrix.GetLength(1) - 1, len = matrix.GetLength(0);
        while (row < len && col >= 0)
        {
            if (matrix[row, col] == target) return true;
            if (matrix[row, col] > target) --col;
            else ++row;
        }
        return false;
    }
}

File: python/240. Search a 2D Matrix II.py
# -*- coding: utf-8 -*-

"""
@Author: xiezizhe
@Date: 24/6/2020 下午3:54
"""


class Solution:

    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix or not matrix[0]:
            return False

        row, col = 0, len(matrix[0]) - 1

        while row < len(matrix) and 0 <= col:

            if matrix[row][col] == target:
                return True
            elif matrix[row][col] > target:
                col -= 1
            else:
                row += 1

        return False
